config/env.go

```diff
@@ -52,6 +58,18 @@ const (
 // It will not fail if no or only part of the variables are present.
 func FromEnv() HCPConfigOption {
 	return func(config *hcpConfig) error {
+		// Read user profile information from the environment, the values will only be
```

Comment:
I would move this block to the bottom of `FromEnv()`. The client credentials/authURL checks are more critical to the SDK to function and should fail early, whereas profile variables should always be optional.